qcodes/dataset/sqlite_base.py

```diff
@@ -112,6 +114,47 @@ class ConnectionPlus(wrapt.ObjectProxy):
 SomeConnection = Union[sqlite3.Connection, ConnectionPlus]
 
 
+def upgrader(func: Callable[[SomeConnection], None]):
+    """
+    Decorator for database version upgrade functions. An upgrade function
+    must have the name `perform_db_upgrade_N_to_M` where N = M-1.
```

Comment:
can the full interface of the upgrade function be described here? i mean, the fact that it has a single argument which is sql connection object, the fact that it either returns or raises (see a comment in the code), the fact that the return values are ignored, etc? 

i understand that one can look into an existing upgrader function and deduct this info, but that might be time consuming.

Reply:
Good point. Done.